src/ui/sources/capturer.js

```diff
@@ -0,0 +1,83 @@
+import toast from 'cogo-toast';
+
+export function stopAudioCapture(stream, dispatch) {
+  if (stream) {
+    stream.getAudioTracks().forEach(track => track.stop());
+    const streamWithoutAudio = new MediaStream([...stream.getVideoTracks()]);
+    dispatch({ type: 'SHARE_DISPLAY', payload: streamWithoutAudio });
+  }
+}
+
+export async function startAudioCapture(stream, dispatch, deviceId = null) {
+  try {
+    const audioStream = await navigator.mediaDevices.getUserMedia({
+      audio: deviceId ? { deviceId } : true,
+      video: false
+    });
+
+    const streamWithAudio = new MediaStream([
+      ...stream.getVideoTracks(),
+      ...audioStream.getAudioTracks()
+    ]);
+
+    dispatch({ type: 'SHARE_DISPLAY', payload: streamWithAudio });
+  } catch (err) {
+    // TODO: there several types of exceptions; certainly we should differentiate here one day
+    console.error('Error: ' + err);
+
+    // TODO: l10n!
+    toast.error('Your browser does not allow to capture audio.');
+
+    stopAudioCapture(stream, dispatch);
+  }
+}
+
+export async function startDisplayCapture(dispatch, displayMediaOptions) {
+  try {
+    const stream = await navigator.mediaDevices.getDisplayMedia(displayMediaOptions);
+    dispatch({ type: 'SHARE_DISPLAY', payload: stream });
+  } catch (err) {
+    // TODO: there 7 types of exceptions; certainly we should differentiate here one day
+    console.error('Error: ' + err);
+
+    // TODO: l10n!
+    toast.error('Your browser does not allow to capture the display.');
+
+    dispatch({ type: 'UNSHARE_DISPLAY' });
+  }
+}
+
+const defaultUserconstraints = {
```

Comment:
Shouldn't this be `defaultUserConstraints`?

Reply:
Nice catch!